4.analyze_data/IoU.ipynb

```diff
@@ -0,0 +1,78 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Perform Intersection over Union (IoU) analysis on object masks "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# iou function from stackoverflow user gil.fernandes\n",
+    "# https://stackoverflow.com/questions/66595055/fastest-way-of-computing-binary-mask-iou-with-numpy\n",
+    "\n",
+    "import torch\n",
+    "import cv2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "tensor(1.)\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Intersection over Union function (IoU) utilizing ground truth (gt) and test binary masks \n",
+    "def iou(gt, test):\n",
```

Comment:
consider adding a docstring here and type hints

Reply:
Got it :) Thanks